subxt/src/error/mod.rs

```diff
@@ -19,7 +19,8 @@ pub use dispatch_error::{
 
 // Re-expose the errors we use from other crates here:
 pub use crate::metadata::Metadata;
-pub use jsonrpsee_types::error::ErrorObject;
+#[cfg(feature = "jsonrpsee")]
+pub use jsonrpsee::types::error::ErrorObject;
```

Comment:
I'd use the cfg macro we have like:

```
crate::macros::cfg_jsonrpsee! {
    // Things hidden by feature
}
```

But I also wonder whether the error you want is `jsonrpsee::core::client::Error`, because `ErrorObject`, which I think isn't very useful for downcasting etc?

Reply:
maybe we can just re-export full crate here instead of jsonrpsee::types::...

Reply:
Yeah, you are correct it would be better to use `pub use jsonrpsee::core::client::Error as JsonrpseeClientError`. To support all errors that the jsonrpsee client can emit.

Maybe it works to downcast it to a ErrorObject if the error [is this variant](https://github.com/paritytech/jsonrpsee/blob/master/core/src/client/error.rs#L36-#L38)

Reply:
> Maybe it works to downcast it to a ErrorObject if the error [is this variant](https://github.com/paritytech/jsonrpsee/blob/master/core/src/client/error.rs#L36-#L38)

downcasting basically just takes some `Box<dyn Error>` and tries to turn it into the concrete type, and for us that type is `jsonrpsee::core::client::Error` I think, so it would never work to downcast to `ErrorObject` unless I'm missing something, I think!

Reply:
ah let me try maybe I was casting to the wrong type

> Maybe it works to downcast it to a ErrorObject if the error is this variant

I am probably hitting this nested variant